lib/checkfunctions.cpp

```diff
@@ -380,7 +380,7 @@ static const Token *checkMissingReturnScope(const Token *tok, const Library &lib
                     else if (switchToken->str() == "{" && (switchToken->scope()->isLoopScope() || switchToken->scope()->type == Scope::ScopeType::eSwitch))
                         switchToken = switchToken->link();
                 }
-                if (!hasDefault)
+                if (!isExhaustiveSwitch(tok->link()) && !hasDefault)
```

Comment:
the hasDefault here seems redundant, isExhaustiveSwitch will return true if there is a default right.